app/model/sdl/VehicleInfoModel.js

```diff
@@ -258,7 +258,50 @@ SDL.SDLVehicleInfoModel = Em.Object.create(
         'driverDoorAjar': true,
         'passengerDoorAjar': true,
         'rearLeftDoorAjar': true,
-        'rearRightDoorAjar': true
+        'rearRightDoorAjar': true,
+        'doorStatuses': [
```

Comment:
@LitvinenkoIra unfortuantely, all three properties which you want to add are grid-based params. That means you have to correlate them with Multiple Modules feature design and fill them dynamically according to selected emulation.
I suggest you to declare just a three empty arrays here.
Next, please take a look at my commit [here](https://github.com/LuxoftSDL/sdl_hmi/pull/48/commits/53de551da4c41029b089a68b0bd140e89137405c) as an example of how to dynamically fill these arrays. In your case you need additionally calculate col/row/level span according to proposal requirements:

> For Door and Gate, location would utilize Grid to locate/span the actual location of the Door or Gate while status provides appropriate status (CLOSED/LOCKED/AJAR/REMOVED). Roof can be a convertible roof, sunroof/moonroof or simply a removable roof. Based on roof type, parameters location, state and status need to provide appropriate values

So I propose to do the following:
1. When `no emulation` selected, init all three params as an empty arrays
2. When 2x3 or 3x3 selected, for doors use left and right boundaries of vehicle representation to locate them (just take a look at `windowStatus` parameter which has exact the same logic for locating vehicle windows). All of them should be `CLOSED`. This is a dynamic-size element array as a result.
3. When 2x3 or 3x3 selected, for gates, use { 'col' : 0, 'row': <vehicle min row index - 1>, 'level': 0, 'colspan': <vehicle max col index> } for specifying the hood location. And use { 'col' : 0, 'row': <vehicle max row index + 1>, 'level': 0, 'colspan': <vehicle max col index> } for specifying trunk location. Both `CLOSED`. P.S. use `getVehicleMaxIndex()` helper function to calculate max index. Assume that min index is always 0 so I believe it is fine to have row = -1 for hood according to API. This is going to be a 2-element array as a result.
4. When 2x3 or 3x3 selected, for roof, let's assume that we have an entire roof covering entire vehicle. Use `createFullCoverage()` helper function to generate a location element for roof. It will have `CLOSED` state. This is going to be a single-element array as a result.

Reply:
@AKalinich-Luxoft thank you for the review, please check 190f6e1